crypto/ffc/ffc_key_generate.c

```diff
@@ -25,11 +25,11 @@ int ossl_ffc_generate_private_key(BN_CTX *ctx, const FFC_PARAMS *params,
     int ret = 0, qbits = BN_num_bits(params->q);
     BIGNUM *m, *two_powN = NULL;
 
-    /* Deal with the edge case where the value of N is not set */
-    if (N == 0)
-        N = qbits;
+    /* Deal with the edge cases where the value of N and/or s is not set */
     if (s == 0)
-        s = N / 2;
+        goto err;
+    if (N == 0)
+        N = params->keylength ? params->keylength : 2 * s;
```

Comment:
I'm not sure supporting N == 0 is a good idea

Reply:
I still prefer that we just return an error in case of N == 0. As far as I know, this is an internal only function that's never called with N == 0, except in the tests.

Reply:
We use this case! For the known prime parameters when the user does not set the dh->length, it will be 0. And the N will fall back to params->keylength this way.